src/jit_channel/channel_manager.rs

```diff
@@ -0,0 +1,1119 @@
+						err: format!(
+							"Received get_info error for an unknown request: {:?}",
+							request_id
+						),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					})?;
+
+				let _jit_channel = peer_state
+					.inbound_channels_by_id
+					.remove(&jit_channel_id)
+					.ok_or(LightningError {
+						err: format!(
+							"Received get_info error for an unknown channel: {:?}",
+							jit_channel_id
+						),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					})?;
+				Ok(())
+			}
+			None => {
+				return Err(LightningError { err: format!("Received error response for a get_info request from an unknown counterparty ({:?})",counterparty_node_id), action: ErrorAction::IgnoreAndLog(Level::Info)});
+			}
+		}
+	}
+
+	fn handle_buy_request(
+		&self, request_id: RequestId, counterparty_node_id: &PublicKey, params: BuyRequest,
+	) -> Result<(), LightningError> {
+		// TODO: need to perform check on `params.version`.
+		// TODO: if payment_size_msat is specified, make sure opening_fee is >= payment_size_msat.
+		// TODO: if payment_size_msat is specified, make sure opening_fee does not hit overflow error.
+		// TODO: if payment_size_msat is specified, make sure our node has sufficient incoming liquidity from public network to receive it.
+
+		if params.opening_fee_params.is_valid(&self.promise_secret) {
+			let mut outer_state_lock = self.per_peer_state.write().unwrap();
+			let inner_state_lock = outer_state_lock
+				.entry(*counterparty_node_id)
+				.or_insert(Mutex::new(PeerState::default()));
+			let peer_state = inner_state_lock.get_mut().unwrap();
+			peer_state.pending_requests.insert(request_id.clone(), Request::Buy(params.clone()));
+
+			self.enqueue_event(Event::LSPS2(super::event::Event::BuyRequest {
+				request_id,
+				version: params.version,
+				counterparty_node_id: *counterparty_node_id,
+				opening_fee_params: params.opening_fee_params,
+				payment_size_msat: params.payment_size_msat,
+			}));
+		}
+		Ok(())
+	}
+
+	fn handle_buy_response(
+		&self, request_id: RequestId, counterparty_node_id: &PublicKey, result: BuyResponse,
+	) -> Result<(), LightningError> {
+		let outer_state_lock = self.per_peer_state.read().unwrap();
+		match outer_state_lock.get(counterparty_node_id) {
+			Some(inner_state_lock) => {
+				let mut peer_state = inner_state_lock.lock().unwrap();
+
+				let jit_channel_id =
+					peer_state.request_to_cid.remove(&request_id).ok_or(LightningError {
+						err: format!(
+							"Received buy response for an unknown request: {:?}",
+							request_id
+						),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					})?;
+
+				let jit_channel = peer_state
+					.inbound_channels_by_id
+					.get_mut(&jit_channel_id)
+					.ok_or(LightningError {
+						err: format!(
+							"Received buy response for an unknown channel: {:?}",
+							jit_channel_id
+						),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					})?;
+
+				if let Err(e) = jit_channel.invoice_params_received(
```

Comment:
hm not quite because we want to early return on error. i can't really move it into a map_err(...)? either because of the borrowing peer_state mutably twice.